include/psim/core/state_field_valued.hpp

```diff
@@ -0,0 +1,105 @@
+//
+// MIT License
+//
+// Copyright (c) 2020 Pathfinder for Autonomous Navigation (PAN)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+
+/** @file psim/core/state_field_valued.hpp
+ *  @author Kyle Krol
+ */
+
+#ifndef PSIM_CORE_STATE_FIELD_VALUED_HPP_
+#define PSIM_CORE_STATE_FIELD_VALUED_HPP_
+
+#include "state_field.hpp"
+#include "state_field_writable.hpp"
+
+#include <utility>
+
+namespace psim {
+
+/** @brief Writable state field implemenation directly backed by a value.
+ *
+ *  @tparam Underlying type.
+ */
+template <typename T>
+class StateFieldValued : public StateFieldWritable<T> {
+ private:
+  /** @brief Underlying value.
+   */
+  T mutable _value;
```

Comment:
Same here. Mutable is a bad idea--for such base-level objects I feel like we should be very deliberate about the class contracts, and add getters/setters rather than trying to use clever language features to make things less verbose.

Reply:
This one here should definitely not be mutable though, you're correct!